src/rgw/rgw_op.cc

```diff
@@ -7319,6 +7319,12 @@ void RGWDeleteMultiObj::execute(optional_yield y)
     return;
   }
 
+  if (multi_delete->objects.empty()) {
+    s->err.message = "The XML you provided was not well-formed or did not validate against our published schema";
```

Comment:
why not a more specific error message about the empty Objects array?

Reply:
Well, some sdks/clis lookup for the error message as well (e.g. https://github.com/aws/aws-cli/pull/8676). i thought it might be a good idea to keep it consistent with AWS as much as we can, but I guess in this case it doesn't much add any value. I'll change it to a more specific message.

Reply:
> e.g. [aws/aws-cli#8676](https://github.com/aws/aws-cli/pull/8676)

argh, we broke that in https://tracker.ceph.com/issues/59433 and still haven't fixed

Reply:
yeah, I guess we can define some default error messages based on the error codes and render if the message is empty.